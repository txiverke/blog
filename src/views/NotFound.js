// @flow

import React from 'react'
import Helmet from 'react-helmet'
import { Link } from 'react-router-dom'

type Props = {
  match: Object,
  location: Object,
  history: Object,
  staticContext: any
}

const NotFound = (props: Props) => {
  const url = props.location.pathname;
  return (
    <section className="app-view app-view-centered">
      <Helmet title="Page not found" />
      <h1 className="app-content-column">
        <span className="txt-huge">404</span><br />
        <span className="txt">
          Sorry, but the url: 
          <span className="txt-highlight">{` xaviervila.tech${url}  `}</span> 
          does not exist.</span><br />
        <Link className="btn mt10" to="/">Back to the landing page</Link>
      </h1>
    </section>
  )
}

export default NotFound