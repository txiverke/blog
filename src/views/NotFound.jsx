// @flow

import React from 'react'
import { connect } from 'react-redux'
import Helmet from 'react-helmet'
import { Link } from 'react-router-dom'

import { getDictionary } from '../utils/dictionary'

type Props = {
  location: Object,
  language: Object
}

const NotFound = (props: Props) => {
  const url = props.location.pathname
  const DIC = getDictionary(props.language.current)

  return (
    <section className="app-view app-view-centered">
      <Helmet title={DIC.PAGE_NOT_FOUND} />
      <h1 className="app-content-column">
        <span className="txt-huge">404</span><br />
        <span className="txt">
          {DIC.SORRY}  
          <span className="txt-highlight">&nbsp;{` xaviervila.tech${url}  `}&nbsp;</span> 
          {DIC.NOT_EXIST}</span><br />
        <Link className="btn mt10" to="/">{DIC.BACK_TO}</Link>
      </h1>
    </section>
  )
}

const mapStateToProps = state => ({ language: state.language })

export const Unwrapped = NotFound
export default connect(mapStateToProps)(NotFound)