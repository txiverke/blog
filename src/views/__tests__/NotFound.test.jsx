import React from 'react'
import { Unwrapped as UnwrappedNotFound } from '../NotFound'
import { shallow } from 'enzyme'

describe('<NotFound />', () => {
  it('should render itself', () => {
    const language = { current: 'eng' }
    const location = { pathname: 'test-path' }
    const component = shallow(<UnwrappedNotFound language={language} location={location} />)
    expect(component).toMatchSnapshot()
  })
})