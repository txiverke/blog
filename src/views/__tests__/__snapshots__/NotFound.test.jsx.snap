// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound
    language={
        Object {
            "current": "eng",
          }
    }
    location={
        Object {
            "pathname": "test-path",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
          title="Page not found"
/>,
        <h1
          className="app-content-column"
>
          <span
                    className="txt-huge"
          >
                    404
          </span>
          <br />
          <span
                    className="txt"
          >
                    Sorry, but the url: 
                    <span
                              className="txt-highlight"
                    >
                               
                               xaviervila.techtest-path  
                               
                    </span>
                     does not exist.
          </span>
          <br />
          <Link
                    className="btn mt10"
                    replace={false}
                    to="/"
          >
                    Back to the landing page
          </Link>
</h1>,
      ],
      "className": "app-view app-view-centered",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defer": true,
          "encodeSpecialCharacters": true,
          "title": "Page not found",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="txt-huge"
>
              404
</span>,
            <br />,
            <span
              className="txt"
>
              Sorry, but the url: 
              <span
                            className="txt-highlight"
              >
                             
                             xaviervila.techtest-path  
                             
              </span>
               does not exist.
</span>,
            <br />,
            <Link
              className="btn mt10"
              replace={false}
              to="/"
>
              Back to the landing page
</Link>,
          ],
          "className": "app-content-column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "404",
              "className": "txt-huge",
            },
            "ref": null,
            "rendered": "404",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Sorry, but the url: ",
                <span
                  className="txt-highlight"
>
                   
                   xaviervila.techtest-path  
                   
</span>,
                " does not exist.",
              ],
              "className": "txt",
            },
            "ref": null,
            "rendered": Array [
              "Sorry, but the url: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    " xaviervila.techtest-path  ",
                    " ",
                  ],
                  "className": "txt-highlight",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  " xaviervila.techtest-path  ",
                  " ",
                ],
                "type": "span",
              },
              " does not exist.",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Back to the landing page",
              "className": "btn mt10",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Back to the landing page",
            "type": [Function],
          },
        ],
        "type": "h1",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
            title="Page not found"
/>,
          <h1
            className="app-content-column"
>
            <span
                        className="txt-huge"
            >
                        404
            </span>
            <br />
            <span
                        className="txt"
            >
                        Sorry, but the url: 
                        <span
                                    className="txt-highlight"
                        >
                                     
                                     xaviervila.techtest-path  
                                     
                        </span>
                         does not exist.
            </span>
            <br />
            <Link
                        className="btn mt10"
                        replace={false}
                        to="/"
            >
                        Back to the landing page
            </Link>
</h1>,
        ],
        "className": "app-view app-view-centered",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defer": true,
            "encodeSpecialCharacters": true,
            "title": "Page not found",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="txt-huge"
>
                404
</span>,
              <br />,
              <span
                className="txt"
>
                Sorry, but the url: 
                <span
                                className="txt-highlight"
                >
                                 
                                 xaviervila.techtest-path  
                                 
                </span>
                 does not exist.
</span>,
              <br />,
              <Link
                className="btn mt10"
                replace={false}
                to="/"
>
                Back to the landing page
</Link>,
            ],
            "className": "app-content-column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "404",
                "className": "txt-huge",
              },
              "ref": null,
              "rendered": "404",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Sorry, but the url: ",
                  <span
                    className="txt-highlight"
>
                     
                     xaviervila.techtest-path  
                     
</span>,
                  " does not exist.",
                ],
                "className": "txt",
              },
              "ref": null,
              "rendered": Array [
                "Sorry, but the url: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      " xaviervila.techtest-path  ",
                      " ",
                    ],
                    "className": "txt-highlight",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    " xaviervila.techtest-path  ",
                    " ",
                  ],
                  "type": "span",
                },
                " does not exist.",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Back to the landing page",
                "className": "btn mt10",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Back to the landing page",
              "type": [Function],
            },
          ],
          "type": "h1",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
